### PR TITLE
Normalise account identifier row parsing

### PR DESCRIPTION
**Pull request by ushkarev**
Cherry-picked commits from #36 by @forforeach

---

## Intro

Following the BAI2 specifications, the account identifier and summary status sections are compound of two types of components: a common component (customer account number and currency) and summary components (type code, amount, item count, and funds type). Summary components are repeated `n` times. 

## Problem

Some banks emit the last comma at the end of the line and some of them don't. The current parsing mechanism adds a comma by the end of the line in any case. This creates discrepancies and some files are failed to parse because of the redundant comma.

## Proposed solution

When parsing the row into a record, we should normalize the expected amount of commas. We need to be aware of the record type we are working with. We already have the record type, so it is easy. To make the code easier to maintain, I introduced the parsing factory that will map the record type to the appropriate parser. 

Currently, I only implemented the specific parser for the account identifier record (03), but it allows the implementation of normalization for other records easily as needed.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -116,6 +116,9 @@ Remember to update `History`_.
 History
 -------
 
+Unreleased 0.10.0
+    Improve parsing of account identifier records with respect to varying number of commans used by different banks.
```
Comment (ushkarev) [inline README.rst]: @forforeach would you say this is a fair description of the change you’re proposing?

Comment (forforeach) [reply README.rst]: Yes, sure that sounds great

**Review by forforeach — APPROVED** [review]
Thanks!

**Comment by ushkarev** [comment]
[Released on PyPI!](https://pypi.org/project/bai2/0.10.0/)

**Comment by forforeach** [comment]
🙏

**Pull request merged**